docs/manual/commands/interfaces.rst

```diff
@@ -0,0 +1,165 @@
+.. _scripting-interfaces:
+
+==========
+Interfaces
+==========
+
+Introduction
+============
+
+This page provides an overview of the various interfaces available to interact with Qtile's
+command graph.
+
+- ``lazy`` calls
+- when running ``qtile shell``
+- when running ``qtile cmd-obj``
+- when using ``CommandClient`` or ``InteractiveCommandClient`` in python 
+
+The way that these commands are called varies depending on which option you select. However, all
+interfaces follow the same, basic approach: navigate to the desired object and then execute a command
+on that object. The following examples illustrate this principle by showing how the same command can
+be accessed by the various interfaces:
+
+.. code:: bash
+
+    Lazy call:
+    lazy.widget["volume"].increase_volume()
+
+    qtile shell:
+    > cd widget/volume
+    widget[volume] > increase_volume()
+
+    qtile cmd-obj:
+    qtile cmd-obj -o widget volume -f increase_volume
+
+    CommandClient:
+    >>> from libqtile.command.client import CommandClient
+    >>> c = CommandClient()
+    >>> c.navigate("widget", "volume").call("increase_volume")
+
+    InteractiveCommandClient:
+    >>> from libqtile.command.client import InteractiveCommandClient
+    >>> c = InteractiveCommandClient()
+    >>> c.widget["volume"].increase_volume()  
+
+
+The Interfaces
+==============
+
+From the examples above, you can see that there are five main interfaces which
+can be used to interact with Qtile's command graph. Which one you choose will depend
+on how you intend to use it as each interface is suited to different scenarios.
+
+- The ``lazy`` interface is used in config scripts to bind commands to keys and
+  mouse callbacks.
+- The ``qtile shell`` is a tool for exploring the graph my presenting it as a
+  file structure. It is not designed to be used for scripting.
+- For users creating shell scripts, the ``qtile cmd-obj`` interface would be
+  the recommended choice.
+- For users wanting to control Qtile from a python script, there are two available
+  interfaces ``libqtile.command.client.CommandClient`` and
+  ``libqtile.command.client.InteractiveCommandClient``. Users are advised to use the
+  ``InteractiveCommandClient`` as this simplifies the syntax for navigating the graph
+  and calling commands.
```

Comment:
Also here

Reply:
If this is referring to bullets, it builds ok for me locally.